front/lib/api/assistant/configuration/table_query.ts

```diff
@@ -77,3 +87,90 @@ export async function fetchTableQueryActionConfigurations({
 
   return actionsByConfigurationId;
 }
+
+export async function createTableDataSourceConfiguration(
+  auth: Authenticator,
+  tableConfigurations: TableDataSourceConfiguration[],
+  tablesQueryConfig: AgentTablesQueryConfiguration,
+  t: Transaction
+) {
+  // Although we have the capability to support multiple workspaces,
+  // currently, we only support one workspace, which is the one the user is in.
+  // This allows us to use the current authenticator to fetch resources.
+  const allWorkspaceIds = [
+    ...new Set(tableConfigurations.map((tc) => tc.workspaceId)),
+  ];
+  const hasUniqueAccessibleWorkspace =
+    allWorkspaceIds.length === 1 &&
+    auth.getNonNullableWorkspace().sId === allWorkspaceIds[0];
+
+  assert(
+    hasUniqueAccessibleWorkspace,
+    "Can't create TableDataSourceConfiguration for query tables: Multiple workspaces."
+  );
+
+  const globalVault = await VaultResource.fetchWorkspaceGlobalVault(auth);
```

Comment:
This line without a `TODO(GROUPS_INFRA)` seems incorrect. WHy are we fetching the global vault here? That's a pretty big assumption?

Reply:
It's until the next PR that passes the `dataSourceViewId` when saving the query table action. This will be removed. Adding a `TODO(GROUPS_INFRA)` if you want.